homeassistant/components/homematic/__init__.py

```diff
@@ -671,6 +673,11 @@ def _get_devices(hass, discovery_type, keys, interface):
                 and class_name not in HM_IGNORE_DISCOVERY_NODE_EXCEPTIONS.get(param, [])
             ):
                 continue
+            if discovery_type == DISCOVER_SWITCHES and class_name == 'IPKeySwitchLevel':
+                channels.remove(8)
```

Comment:
I don't like those hard-coded values myself, but I am not sure how to make it more generic; suggestions are very welcome

Reply:
@pvizeli what do you think? Because of the architecture of pyhomematic and the way the component works I currently don't see any other way than this.

Reply:
I need a bit more details how that device work :D

Reply:
The important part of Guillems explanation below is, that the `ELEMENT` property contains all 3 actor-channels. Because of that we would normally get 3 lights __and__ 3 switches. Until now actors have always been of only 1 type. If there are more nodes on other devices, then those aren't actor-nodes.  
Essentially we are not able to let one device be different actor types at once. That's why we just put all channels in `ELEMENT`, and then remove those that don't match in accordance to the current discovery type.